pkg/rehearse/jobs.go

```diff
@@ -222,6 +228,107 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 	return rehearsals
 }
 
+// AddRandomJobsForChangedTemplates finds random jobs from the PR config that are using a specific template with a specific cluster type.
+// So if a template will be changed, find the jobs that are using a template in combination with the `aws`,`openstack`,`gcs` and `libvirt` cluster types.
+func AddRandomJobsForChangedTemplates(templatesWithClusterTypes map[string]sets.String, templates config.CiTemplates, prConfigPresubmits map[string][]prowconfig.Presubmit, loggers Loggers, prNumber int) config.Presubmits {
```

Comment:
Maybe I am missing something, but how is this random?

Reply:
golang maps :) The order is random

Reply:
It is [unspecified](https://golang.org/ref/spec#For_range), not random. The (current) implementation is pseudo-random with the exact intent of making people not rely on the iteration order. We should use a proper randomization method if that is what we need.

Reply:
Well, yea I guess we can call it `pseudo-random` better.

Reply:
pseudo-random is something [else](https://en.wikipedia.org/wiki/Pseudorandomness) already, this is just unspecified :D

Reply:
comment added.

Reply:
There are two issues here that should not be mixed:
- The *current implementation* is not random — we are dealing with computers, so obviously we are talking about pseudo-randomness ("pseudo-pseudo-random" sounded too ridiculous, maybe I should have called it "bad pseudo-randomness"). This is a much smaller problem.
- The *iteration order* is unspecified. This is my biggest issue with this code. Again, if we want (pseudo-)randomness, we should use a proper method. To me, relying on it is fundamentally wrong and deeply ironic, given that whatever "randomness" is used in the iteration order was added to discourage this.

Reply:
Well, we don't really care if we will pick the same job. Since the map keys are not sorted, we will end up to pick a job from a different repo. At the end, we just want to rehearse one of the jobs that using the specific template and cluster_type

Reply:
@bbguimaraes like mentioned on slack, I don't think we even need pseudo-randomness. It just gives you some statistic properties about the distribution, but we don't care about that, we just want to avoid hitting the same job every time, for which this is IMHO sufficient.

Reply:
You are both justifying the first issue, which I already said is not nearly as important as the second.

I don't see how I can move this argument forward with more comments, so — even though I still think this is wrong — I'll say: overruled by 2x1.